### PR TITLE
Sample tests for Magnet project

### PR DESCRIPTION
**Pull request by furknyavuz**
Hi @premnirmal,

This pull request is sample for the unit tests that we plan to write in order to raise the overall coverage of the project to around 80%.

Please let me know if you have any questions.

Furkan Yavuz
DevFactory - Code Quality Team

### CODE REVIEW

File: library/src/test/java/com/premnirmal/Magnet/SimpleAnimatorTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        verify(view).startAnimation(animation);
+    }
+
+    @Test
+    public void testStartAnimation() throws Exception {
+        // when
+        simpleAnimator.startAnimation();
+
+        // then
+        verify(animation, never()).setAnimationListener(listener);
```
Comment (premnirmal) [inline library/src/test/java/com/premnirmal/Magnet/SimpleAnimatorTest.java]: how does this verify?

**Comment by premnirmal** [comment]
Thanks! I see the test cases don't subclass junits test case. Do the tests still run?

File: library/src/test/java/com/premnirmal/Magnet/SimpleAnimatorTest.java (hunk 1 above)
Comment (furknyavuz) [inline library/src/test/java/com/premnirmal/Magnet/SimpleAnimatorTest.java]: Hi @premnirmal ,
At this line, we are trying to verify that animation didn't set it's listener.

When we call startAnimation method without listener, animation will never set it' listener. This is the only difference I see.

**Comment by furknyavuz** [comment]
You'r welcome! :)

About the extending from test case, let me double check. Also we can extend test classes from test case at coming commits.

**Comment by furknyavuz** [comment]
Hi @premnirmal,
Could you able to review my comments? If you think it is okay, please accept the pull request, then we can start for working on full test coverage.

Thanks,

**Comment by premnirmal** [comment]
Hi @furknyavuz . Yes it looks good. Do you mind making a new PR that merges your branch into the branch `testing`? I don't want to merge this into master until I set up circleCI to run tests

**Comment by furknyavuz** [comment]
Sure @premnirmal,
I'll open new PR to testing branch.

Thank you,

**Pull request closed without merging**